### ISSUE TITLE
Incorrect issues/PRs count if response from GitHub does not contain expected header

### ISSUE THREAD

**Issue by Jericho** [cake-contrib/Cake.AddinDiscoverer#79]
In the `GetGithubStats` step we attempt to determine how many issues and PRs are currently open in a given GitHub repository. We issue a search query to GitHub and we inspect the headers in the response to derive the information we need. However, this logic does not work if the expected headers are missing.

It appears that GitHub omits the expected headers when all records matching the search criteria can fit on a single page and since our search query specifies only one record per page, this means the headers are omitted when only one record matches our criteria.

Long story short: I need to improve the `GetRecordsCount` method to handle the situation where headers are missing.
